src/Vts/IO/CustomBinaryStreamWriterOfT.cs

```diff
@@ -72,6 +72,9 @@ private void OpenStream()
         {
             try
             {
+                // guard against directory not existing ahead of time
+                Directory.CreateDirectory(Path.GetDirectoryName(_filename));

```

Comment:
Possible 'null' assignment to non-nullable entity

Reply:
Thanks - this one seems maybe a little out of scope - null assignment is rampant across the solution, and I'm just updating a method here where `_filename` is used in the same fashion. But I went ahead and enabled null type checking for the file and updated the code accordingly. Guess it doesn't hurt to update a whole file when we touch it.

Reply:
Since we started cleaning up the code, we are making a point to correct a file if we work on it. Due to the size of code base, it's hard to get to everything unless we are working on it. There are 2 more suggestions at the top of this file, to make _filename and _writeMap readonly.

Reply:
I thought I had addressed those in my changeset but can double check in the morning

Reply:
The readonly is good now, it's just the IDisposable implementation that's left.